SpatialGDK/Source/SpatialGDK/Private/Tests/RPCServiceTest.cpp

```diff
@@ -0,0 +1,549 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "CoreMinimal.h"
+#include "Interop/SpatialRPCService.h"
+#include "Interop/SpatialStaticComponentView.h"
+#include "Schema/RPCPayload.h"
+#include "SpatialConstants.h"
+#include "SpatialGDKSettings.h"
+#include "Tests/TestDefinitions.h"
+#include "Utils/RPCRingBuffer.h"
+
+#define RPC_SERVICE_TEST(TestName) \
+	GDK_TEST(Core, SpatialRPCService, TestName)
+
+// Test Globals
+namespace
+{
+	enum ERPCEndpointType : uint8_t
+	{
+		INVALID,
+		SERVER_AUTH,
+		CLIENT_AUTH,
+		SERVER_AND_CLIENT_AUTH,
+		NO_AUTH
+	};
+
+	struct EntityPayload
+	{
+		EntityPayload(Worker_EntityId InEntityID, const SpatialGDK::RPCPayload& InPayload)
+			: EntityId(InEntityID)
+			, Payload(InPayload)
+		{}
+
+		Worker_EntityId EntityId;
+		SpatialGDK::RPCPayload Payload;
+	};
+
+	Worker_EntityId RPCTestEntityId_1 = 1;
+	Worker_EntityId RPCTestEntityId_2 = 24;
+	Worker_EntityId RPCTestEntityId_3 = 65;
+	Worker_EntityId RPCTestEntityId_4 = 4;
+	Worker_EntityId RPCTestEntityId_5 = 100;
+
+	SpatialGDK::RPCPayload SimplePayload = SpatialGDK::RPCPayload(1, 0, TArray<uint8>({ 1 }, 1));
+
+	ExtractRPCDelegate DefaultRPCDelegate = ExtractRPCDelegate::CreateLambda([](Worker_EntityId EntityId, ERPCType RPCType, const SpatialGDK::RPCPayload& Payload) {
+		return true;
+	});
+
+	void AddEntityComponentToStaticComponentView(USpatialStaticComponentView& StaticComponentView,
+		const SpatialGDK::EntityComponentId& EntityComponentId,
+		Schema_ComponentData* ComponentData,
+		Worker_Authority Authority)
+	{
+		Worker_AddComponentOp AddComponentOp;
+		AddComponentOp.entity_id = EntityComponentId.EntityId;
+		AddComponentOp.data.component_id = EntityComponentId.ComponentId;
+		AddComponentOp.data.schema_type = ComponentData;
+		StaticComponentView.OnAddComponent(AddComponentOp);
+
+		Worker_AuthorityChangeOp AuthorityChangeOp;
+		AuthorityChangeOp.entity_id = EntityComponentId.EntityId;
+		AuthorityChangeOp.component_id = EntityComponentId.ComponentId;
+		AuthorityChangeOp.authority = Authority;
+		StaticComponentView.OnAuthorityChange(AuthorityChangeOp);
+	}
+
+	void AddEntityComponentToStaticComponentView(USpatialStaticComponentView& StaticComponentView, const SpatialGDK::EntityComponentId& EntityComponentId, Worker_Authority Authority)
+	{
+		AddEntityComponentToStaticComponentView(StaticComponentView, EntityComponentId, Schema_CreateComponentData(), Authority);
+	}
+
+	Worker_Authority GetClientAuthorityFromRPCEndpointType(ERPCEndpointType RPCEndpointType)
+	{
+		switch (RPCEndpointType)
+		{
+		case CLIENT_AUTH:
+		case SERVER_AND_CLIENT_AUTH:
+			return WORKER_AUTHORITY_AUTHORITATIVE;
+			break;
+		case SERVER_AUTH:
+		default:
+			return WORKER_AUTHORITY_NOT_AUTHORITATIVE;
+			break;
+		}
+	}
+
+	Worker_Authority GetServerAuthorityFromRPCEndpointType(ERPCEndpointType RPCEndpointType)
```

Comment:
Is it intentional, that `GetClientAuthorityFromRPCEndpointType` and `GetServerAuthorityFromRPCEndpointType` are only called with `SERVER_AUTH` argument?

Reply:
Also called as part of AddEntityToStaticComponentView where RPCEndpointType can be any

Reply:
Ah, right, resolving.